markdown/markdown.go

```diff
@@ -266,11 +267,132 @@ func (r *renderer) RenderNode(ctx context.Context, w io.Writer, node *blackfrida
 				}
 				return blackfriday.GoToNext
 			}
+			if newNodes := rewriteDates(node); len(newNodes) > 0 {
+				for _, n := range newNodes {
+					if status := r.Renderer.RenderNode(w, n, entering); status != blackfriday.GoToNext {
+						return status
+					}
+				}
+				return blackfriday.GoToNext
+			}
 		}
 	}
 	return r.Renderer.RenderNode(w, node, entering)
 }
 
+var datePattern = regexp.MustCompile(
+	// For examples, see:
+	// https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time#valid_datetime_values
+	// https://developer.mozilla.org/en-US/docs/Web/HTML/Date_and_time_formats
+	`\b(` +
+		// Month string: MM-DD
+		`[0-1][1-9]-[0-3][1-9]|` +
+		// Week string: YYYY-WXX
+		`\d{4}-W\d?\d|` +
+		// Date or datetime string, with optional timezone: YYYY-MM-DD HH:MM:SS+HH:MM
+		`\d{4}-\d{2}(-\d{2}([ T]?[0-2]?\d:\d\d(:\d\d(\.\d+)?)?(Z|[+-]\d\d:?\d\d)?)?)?|` +
+		// Time string, with optional timezone: HH:MM:SS+HH:MM+HH:MM
+		`(T?[0-2]?\d:\d\d(:\d\d(\.\d+)?)?(Z|[+-]\d\d:?\d\d)?)|` +
+		// Fiscal year with optional quarter
+		`FY[' -]?(\d{2,4})(?:[' -]?F?Q([1-4]))?|` +
+		// Fiscal quarter only
+		`F?Q([1-4])` +
+		`)\b`)
+
+var fiscalIntervalPattern = regexp.MustCompile(
+	`^(?:` +
+		// Fiscal year
+		`FY[' -]?(\d{2,4})(?:[' -]?F?Q([1-4]))?|` +
+		// Fiscal quarter only
+		`F?Q([1-4])` +
+		`)$`)
+
+// parseFiscalInterval returns a YYYY-MM-DD or MM-DD date string for a reference to a fiscal quarter or fiscal year,
+// for the start of the fiscal quarter or fiscal year, respectively.
+func parseFiscalInterval(dateStr string) string {
+	matches := fiscalIntervalPattern.FindStringSubmatch(dateStr)
+	if len(matches) == 0 {
+		return ""
+	}
+	fyStr := matches[1]
```

Comment:
You probably want to confirm that there are enough matches before indexing into `matches` or it'll panic.

Reply:
There will always be the same number of subexpressions if there is a match (which is checked by the `len(matches) == 0` above), so this _should_ never fail, but I think that could be made more clear with named subexpressions

Reply:
I looked at named subexpressions but looks like in Go named capture groups are accessed by mapping them to a numeric index in a loop, which just seemed like extra steps compared to assigning them to variables: https://stackoverflow.com/a/20751656/4208018
It's also not possible to have the same name for two capture groups, so there would still need to be some logic to map the 2nd and 3rd capture group to the same variable.

Reply:
Cool, not a big deal. Thanks for looking into it 👍